src/backend/api/mirrors_update.py

```diff
@@ -0,0 +1,236 @@
+            headers=HEADERS,
+        )
+        request.raise_for_status()
+    except requests.RequestException:
+        logger.error(
+            'Mirror "%s" has no timestamp file by url "%s"',
+            mirror_info['name'],
+            timestamp_url,
+        )
+        mirror_info['status'] = 'expired'
+        return
+    try:
+        mirror_should_updated_at = dateparser.parse(
+            f'now-{allowed_outdate} UTC'
+        ).timestamp()
+        mirror_last_updated = float(request.content)
+        if mirror_last_updated > mirror_should_updated_at:
+            mirror_info['status'] = 'ok'
+        else:
+            mirror_info['status'] = 'expired'
+        return
+    except AttributeError:
+        mirror_info['status'] = 'expired'
+        return
+
+
+def get_verified_mirrors(
+        mirrors_dir: AnyStr,
+        versions: List[AnyStr],
+        repos: List[Dict[AnyStr, Union[Dict, AnyStr]]],
+        allowed_outdate: AnyStr
+) -> List[Dict[AnyStr, Union[Dict, AnyStr]]]:
+    """
+    Loop through the list of mirrors and return only available
+    and not expired mirrors
+    :param mirrors_dir: path to the directory which contains
+           config files of mirrors
+    :param versions: the list of versions which should be provided by mirrors
+    :param repos: the list of repos which should be provided by mirrors
+    :param allowed_outdate: allowed mirror lag
+    """
+
+    result = []
+    for config_path in Path(mirrors_dir).rglob('*.yml'):
+        with open(str(config_path), 'r') as config_file:
+            mirror_info = yaml.safe_load(config_file)
+            if 'name' not in mirror_info:
+                logger.error(
+                    'Mirror file "%s" doesn\'t have name of the mirror',
+                    config_path,
+                )
+                continue
+            if 'address' not in mirror_info:
+                logger.error(
+                    'Mirror file "%s" doesn\'t have addresses of the mirror',
+                    config_path,
+                )
+                continue
+            if mirror_info['name'] in WHITELIST_MIRRORS:
+                mirror_info['status'] = 'ok'
+                set_repo_status(mirror_info, allowed_outdate)
+                set_geo_data(mirror_info)
+                result.append(mirror_info)
+                continue
+            if mirror_available(
+                    mirror_info=mirror_info,
+                    versions=versions,
+                    repos=repos,
+            ):
+                set_repo_status(mirror_info, allowed_outdate)
+                set_geo_data(mirror_info)
+                result.append(mirror_info)
+    return result
+
+
+def set_geo_data(
+        mirror_info: Dict[AnyStr, Union[Dict, AnyStr]],
+) -> None:
+    """
+    Set geo data by IP of a mirror
+    :param mirror_info: Dict with info about a mirror
+    """
+
+    mirror_name = mirror_info['name']
+    ip = socket.gethostbyname(mirror_name)
```

Comment:
Something like missing payment for domain or missing A record may break the whole tool, need some error handling here.

Reply:
It's added try...except